src/Aspire.Hosting.Azure.SignalR/AzureSignalRHealthCheck.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting.Azure.SignalR;
+internal sealed class AzureSignalRHealthCheck : IHealthCheck
+{
+    private readonly HttpClient _client;
+    private readonly String _hostName;
+
+    public AzureSignalRHealthCheck(string hostName)
+    {
+        ArgumentNullException.ThrowIfNull(hostName, nameof(hostName));
+        _client = new HttpClient();
```

Comment:
DI in HttpClient instead, https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-factory#the-ihttpclientfactory-type